### ISSUE TITLE
AD sync not working

### ISSUE THREAD

**Issue by cprinze** [tynx/SMSChallenge#4]
Hello,

the AD-sync ist not working. syncing a single, named user is wokring fine.
how can i debug it?

**Comment by tynx** [comment]
In the config you can configure the logger so that you have the error-messages in a file/db/syslog. that'd be a good start for debugging.

A good guess for you problem may be the configuration of following fields in the config:
```
# where to look for groups
base_ou_for_groups= 
# all the groups comma seperated
groups_to_sync=
```

The syncing of all users depends on these two lines, while this is not true for syncing one user.

**Comment by cprinze** [comment]
maybe we can do a teamviewer session, would be great

**Comment by cprinze** [comment]
base_ou_for_groups=dc=domain,dc=local
groups_to_sync=cn=Benutzer Container
the Benutzer Container name contains a space
is this okay?

**Comment by tynx** [comment]
Could you please provide me the content of the log?
Also the groups_to_sync requires the full path to the group, if I remember correctly.

**Comment by cprinze** [comment]
The only line there is: Jan 27 10:48:05 php: smschallenge_sync: INFO Successfully synced 0 users. There are 1 users which were not synced
it should sync about 400 users.

base_ou_for_groups=dc=domain,dc=local
groups_to_sync=cn=Benutzer Container,dc=domain,dc=local

**Comment by tynx** [comment]
Can you try with:
```
groups_to_sync=Benutzer Container
```
?

**Comment by cprinze** [comment]
same log:
Jan 27 11:03:01 php: smschallenge_sync: INFO Successfully synced 0 users. There are 1 users which were not synced

**Comment by wangibangi** [comment]
Please try the following

base_ou_for_groups=DC=domain,DC=local
groups_to_sync=Benutzer Container

if this does not work, also try this:
base_ou_for_groups=OU=Benutzer Container,DC=domain,DC=local
groups_to_sync=

**Comment by cprinze** [comment]
done the first thing above.

for the second same error:
Jan 27 11:19:50 php: smschallenge_sync: INFO Successfully synced 0 users. There are 1 users which were not synced

**Comment by wangibangi** [comment]
Did the first thing work?

_The error message indicates that the ad sync is working, otherwise it would not recognize the 1 not synced user. Somehow the user is not recognized in the group to sync._
=> THIS IS WRONG

maybe try
base_ou_for_groups=DC=domain,DC=local
groups_to_sync=*

EDIT: not true, I missunderstod it.

**Comment by cprinze** [comment]
with
base_ou_for_groups=OU=Benutzer Container,DC=domain,DC=local
groups_to_sync=*
return:
 php -e sync/sync.php sync
no valid return...

with:
base_ou_for_groups=DC=domain,DC=local
groups_to_sync=*
return:
 php -e sync/sync.php sync
PHP Warning:  ldap_search(): Partial search results returned: Sizelimit exceeded in /opt/smschallenge/lib/Ldap.php on line 211
no valid return...

in the log:
Jan 27 11:47:11 php: smschallenge_sync: INFO Successfully synced 0 users. There are 1 users which were not synced
Jan 27 11:47:54 php: smschallenge_sync: INFO Successfully synced 0 users. There are 1 users which were not synced

**Comment by wangibangi** [comment]
Aaah.

"there are 1 users which were not synced" => this means there was one user in the database which was not synced against the AD via ldap. (probably the one you previously synced via syncuser )

what is the outpout of the php script if you use
base_ou_for_groups=

base_ou_for_groups=DC=domain,DC=local
groups_to_sync=Benutzer Container

Also how many users are in this group?
Its been a while since we developed this.... it seems like the ldap search doesn't support partial search.

**Comment by cprinze** [comment]
yes, i looked up in the database, it is the one synced via syncuser.

the output for "base_ou_for_groups= " is:
php -e sync/sync.php sync
PHP Warning:  ldap_search(): Search: No such object in /opt/smschallenge/lib/Ldap.php on line 21

for the second one it ist nothing
in the group there are about 400 Users.

**Comment by wangibangi** [comment]
Sorry I meant 
"base_ou_for_groups=DC=domain,DC=local
groups_to_sync=Benutzer Container"

maybe try this to edit line 211 in Ldap.php and set a bigger sizelimit.
http://stackoverflow.com/questions/24913366/php-ldap-overcoming-size-limit
http://php.net/manual/de/function.ldap-search.php

**Comment by cprinze** [comment]
hmm, if i do it this way:
base_ou_for_groups=OU=Benutzer Container,DC=domain,DC=local
groups_to_sync=*

the return of "php -e sync/sync.php sync"is:
no valid return...


for the proposed solution i can figure out what the line should look like :(

**Comment by wangibangi** [comment]
Are you sure the group you want to Sync is "Benutzer Container" ?

Can you maybe try to create a group just for testing and try with this one?
Teamviewer would be ok --> send me an email

**Comment by cprinze** [comment]
yes, i am sure as the bind user resides in this container.
ID and Password is sent
